card/mod/05_standard/set/abstract/attachment.rb

```diff
@@ -54,7 +54,10 @@ def assign_attachment file, original_filename
   update_column(:db_content,attachment.db_content(mod: load_from_mod))
   expire
 end
-event :save_original_filename, after: :validate_name, when: proc {|c| !c.preliminary_upload? && !c.save_preliminary_upload? && c.attachment_changed?} do
+event :save_original_filename,
+      after: :validate_name,
+      when: proc {|c| c.attachment.file.present? && !c.preliminary_upload? &&
+                      !c.save_preliminary_upload? && c.attachment_changed?} do
```

Comment:
Expression at 60, 75 should be on its own line.<br>Use 2 (not 16) spaces for indenting an expression spanning multiple lines.